catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -74,15 +87,22 @@ public RemoteSolrCatalogProvider(
       SolrClient client,
       SolrFilterDelegateFactory solrFilterDelegateFactory,
       @Nullable DynamicSchemaResolver resolver) {
-    this.provider =
-        new SolrCatalogProvider(
-            client,
-            filterAdapter,
-            solrFilterDelegateFactory,
-            (resolver == null) ? new DynamicSchemaResolver() : resolver);
+    this.filterAdapter = filterAdapter;
+    this.client = client;
+    this.solrFilterDelegateFactory = solrFilterDelegateFactory;
+    this.resolver = resolver;
+    init();
     provider.maskId(getId());
   }
 
+  protected void init() {
+    if (resolver == null) {
+      resolver = new DynamicSchemaResolver(anyTextAttributes, Collections.emptyList());
```

Comment:
❗️ This should be moved to the constructor to avoid potentially leaving `resolver` to `null` and making it clear that it can never be.